main.tf

```diff
@@ -60,6 +60,12 @@ resource "aws_transfer_ssh_key" "default" {
   ]
 }
 
+resource "aws_eip" "sftp" {
+  count = local.enabled && var.provision_eip ? length(var.subnet_ids) : 0
```

Comment:
wouldn't it be better to use a `for_each` here? So if one of the subnets was removed, it would only remove the eip for the subnet removed instead of having to recreate the ones with the new indices.

Reply:
The above was causing issues `The "for_each" value depends on resource attributes that cannot be determined
until apply` and I spoke with @bradj and he doesn't think that using count instead of for_each would be an issue.